3-plot_category_kappa.R

```diff
@@ -43,28 +41,82 @@ array.files <- lf[grepl(paste0(file_identifier,
+  
+  # check that we have ordered pairs of regular and null files for array and seq
+  array_seeds <- stringr::str_sub(array.files, -8, -5)
+  null_array_seeds <- stringr::str_sub(null_array.files, -8, -5)
+  seq_seeds <- stringr::str_sub(seq.files, -8, -5)
+  null_seq_seeds <- stringr::str_sub(null_seq.files, -8, -5)
+  if (!(all(array_seeds == null_array_seeds) &
+        all(seq_seeds == null_seq_seeds))) {
+    stop("Array or seq seeds do not match in delta kappa plotting script.")
+  }
+  
+}
 
 # define output files
-plot.file.lead <- paste0(file_identifier, "_train_3_models_kappa_")
-summary.df.filename <- file.path(res.dir,
+plot.file.lead <- ifelse(null_model,
+                         paste0(file_identifier, "_train_3_models_delta_kappa_"),
+                         paste0(file_identifier, "_train_3_models_kappa_"))
+summary.df.filename <- ifelse(null_model,
+                              file.path(res.dir,
+                                        paste0(file_identifier,
+                                               "_train_3_models_delta_kappa_summary_table.tsv")),
+                              file.path(res.dir,
                                  paste0(file_identifier,
-                                        "_train_3_models_summary_table.tsv"))
+                                        "_train_3_models_kappa_summary_table.tsv")))
 
 #### read in data --------------------------------------------------------------
 
 # read in the tables that contain the kappa statistics for predictions on test
 # data
 array.list <- list()  # initialize list that will hold all array tables
 seq.list <- list()  # initialize list that will hold all the RNA-seq tables
-for (i in 1:length(array.files)) {
-  array.list[[i]] <- fread(array.files[i], data.table = F)
-  seq.list[[i]] <- fread(seq.files[i], data.table = F)
+for (file_index in 1:length(array.files)) {
+  array.list[[file_index]] <- fread(array.files[file_index], data.table = F)
+  seq.list[[file_index]] <- fread(seq.files[file_index], data.table = F)
+}
+
+if (null_model) {
+  null_array.list <- list() # initialize list that will hold null array tables
+  null_seq.list <- list() # initialize list that will hold null RNA-seq tables
+  for (null_file_index in 1:length(null_array.files)) {
+    null_array.list[[null_file_index]] <- fread(null_array.files[null_file_index], data.table = F)
+    null_seq.list[[null_file_index]] <- fread(null_seq.files[null_file_index], data.table = F)
+  }
+  
+  # calculate delta kappa values
+  delta_kappa_array.list <- list() # list for delta kappa array values
+  delta_kappa_seq.list <- list() # list for delta kappa seq values
+  for (pair_index in 1:length(array.files)) {
+    delta_kappa_array.list[[pair_index]] <- array.list[[pair_index]] %>%
+      left_join(null_array.list[[pair_index]],
+                by = c("perc.seq", "classifier", "norm.method")) %>%
+      mutate(delta_kappa = kappa.x - kappa.y) %>% # regular kappa - null kappa
```

Comment:
I would use `suffix` here so that this can be a little more explicit:
```suggestion
      left_join(null_array.list[[pair_index]],
                by = c("perc.seq", "classifier", "norm.method"),
                suffix = c(".true", ".null")) %>%
      mutate(delta_kappa = kappa.true - kappa.null) %>% # regular kappa - null kappa
```

If you make this change here, don't forget to apply it to the seq step below too!

Reply:
Oooh I like that!